main.py

```diff
@@ -0,0 +1,361 @@
+        headers={"WWW-Authenticate": "Bearer"}
+    )
+
+
+@app.post("/users/", tags=["User"], status_code=status.HTTP_201_CREATED, response_model=user_pydanticOut)
+async def user_registration(user: user_pydanticIn):
+    user_info = user.dict(exclude_unset=True)
+
+    # This is a bad way to do it:
+    # TODO fix this, create custom 'user_pydanticIn for validate data '
+    if len(user_info["password"]) < 8:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="Password must be longer than 8 characters")
+    if len(user_info["username"]) < 5:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="Username must be longer than 5 characters")
+    if is_not_email(user_info["email"]):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="This is not a valid email")
+
+    if await User.exists(username=user_info.get("username")):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="Username already exists")
+    if await User.exists(email=user_info.get("email")):
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST, detail="Email already exists")
+
+    user_info["password"] = get_hashed_password(user_info["password"])
+
+    user_obj = await User.create(**user_info)
+    user_new = await user_pydanticOut.from_tortoise_orm(user_obj)
+    return user_new
+
+
+template = Jinja2Templates(directory="templates")
+
+
+@app.get("/verification/email", response_class=HTMLResponse, tags=["User"])
+async def email_verification(request: Request, token: str):
+    user = await very_token_email(token)
+    if user:
+        if not user.is_verified:
+            user.is_verified = True
+            await user.save()
+        context = {
+            "request": request,
+            "is_verified": user.is_verified,
+            "username": user.username
+        }
+        return template.TemplateResponse("verification.html", context)
+
+    raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Invalid Token or expired token",
+        headers={"WWW-Authenticate": "Bearer"}
+    )
+
+
+@app.post("/uploadfile/profile", tags=["User"])
+async def upload_profile_image(file: UploadFile = File(..., max_lenght=10485760),
+                               user: user_pydantic = Depends(get_current_user)):
+
+    FILEPATH = "./static/images/"
+    file_name = file.filename
+
+    try:
+        extension = file_name.split(".")[1]
+    finally:
+        if extension not in ["png", "jpg", "jpeg"]:
+            raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST,
+                                detail="File extension not allowed")
+
+    token_name = "logo" + secrets.token_hex(10) + "." + extension
+    generated_name = FILEPATH + token_name
+    file_content = await file.read()
+
+    with open(generated_name, "wb") as f:
```

Comment:
**Performance Issue**: Writing files synchronously within an asynchronous endpoint can block the event loop, leading to performance degradation, especially under high load. <br> **Fix**: Use an asynchronous file write operation or offload the file writing to a background task to avoid blocking the event loop. <br> **Code Suggestion**: 
 ```
 - with open(generated_name, "wb") as f:
 -     f.write(file_content)
 + async with aiofiles.open(generated_name, "wb") as f:
 +     await f.write(file_content)
 ```